packagedb/models.py

```diff
@@ -1219,6 +1219,153 @@ def make_relationship(
         relationship=relationship,
     )
 
+class UpdateMixin:
+    """
+    Provide a ``update()`` method to trigger a save() on the object with the
+    ``update_fields`` automatically set to force a SQL UPDATE.
+    """
+
+    def update(self, **kwargs):
+        """
+        Update this resource with the provided ``kwargs`` values.
+        The full ``save()`` process will be triggered, including signals, and the
+        ``update_fields`` is automatically set.
+        """
+        for field_name, value in kwargs.items():
+            setattr(self, field_name, value)
+
+        self.save(update_fields=list(kwargs.keys()))
+
+
+class PackageWatch(models.Model, UpdateMixin):
+    """
+    Model representing a watch on a package to monitor for new versions.
+    """
+
+    DEPTH_CHOICES = (
+        ("version", "Version"),
+        ("metadata", "Metadata"),
+        ("scan", "Scan"),
+    )
+
+    FREQUENCY_CHOICES = (
+        ("daily", "Daily"),
+        ("weekly", "Weekly"),
+        ("monthly", "Monthly"),
+    )
+
+    uuid = models.UUIDField(
+        verbose_name=_("UUID"),
+        default=uuid.uuid4,
+        unique=True,
+        editable=False,
+        help_text=_(
+            'The identifier of the PackageWatch.'
+        )
+    )
+
+    package_url = models.CharField(
+        max_length=2048,
+        unique=True,
+        null=False,
+        blank=False,
+        db_index=True,
+        help_text=_("PackageURL to watch."),
```

Comment:
We should have a "soft delete" field as otherwise we cannot ever deactivate a watch and are forced to delete it.
```Python
    is_active = models.BooleanField(
        null=True,
        db_index=True,
        default=True,
        help_text=("When set to True (Yes), this Package Watch is active. When set to False (No), this watch is inactive and not processed."
        ),
    )
```

Reply:
We should have a "soft delete" field as otherwise we cannot ever deactivate a watch and are forced to delete it.
```Python
    is_active = models.BooleanField(
        null=True,
        db_index=True,
        default=True,
        help_text=("When set to True (Yes), this Package Watch is active. When set to False (No), this watch is inactive and not processed."
        ),
    )
```